libs/traits/src/liquidity_pools.rs

```diff
@@ -23,6 +23,9 @@ use sp_std::vec::Vec;
 pub trait LPEncoding: Sized {
 	fn serialize(&self) -> Vec<u8>;
 	fn deserialize(input: &[u8]) -> Result<Self, DispatchError>;
+
+	fn get_message_proof(&self) -> Option<[u8; 32]>;
```

Comment:
NIT. Can we use a type alias for `[u8; u32]`, maybe called `Proof` or similar? I would add it just before LPEncoding trait, as "part" of the interface

Reply:
OK, was thinking about doing that but wasn't sure if we need extra constraints on it so that we can use it in storage. Gonna try it out.

Reply:
I think by now, we can force the type to be that type. Everything regarding LP is quite tied to our use case. It's not as generic as other pallets.